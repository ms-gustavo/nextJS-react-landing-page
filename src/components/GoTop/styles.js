import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    position: fixed;
    bottom: 2rem;
    z-index: 6;
    right: 2rem;
    background: ${theme.colors.primaryColor};
    color: ${theme.colors.white};
    display: flex;
    align-items: center;
    justify-content: center;
    width: 4rem;
    height: 4rem;
    opacity: 0.7;
    border-radius: 50%;
  `}
`;
