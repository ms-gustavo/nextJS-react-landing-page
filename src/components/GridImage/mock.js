export default {
  background: false,
  title: 'My grid image',
  description:
    'Lorem ipsum dolor sit, amet consectetur adipisicing elit. Provident harum quo esse cupiditate et ab optio debitis, est facilis in deleniti inventore vel aperiam cumque consequuntur nesciunt nemo qui necessitatibus.',
  grid: [
    {
      altText: 'Teste 1',
      srcImg: 'https://source.unsplash.com/random/800x800',
    },
    {
      altText: 'Teste 2',
      srcImg: 'https://source.unsplash.com/random/800x800',
    },
    {
      altText: 'Teste 3',
      srcImg: 'https://source.unsplash.com/random/800x800',
    },
    {
      altText: 'Teste 4',
      srcImg: 'https://source.unsplash.com/random/800x800',
    },
    {
      altText: 'Teste 5',
      srcImg: 'https://source.unsplash.com/random/800x800',
    },
    {
      altText: 'Teste 6',
      srcImg: 'https://source.unsplash.com/random/800x800',
    },
  ],
};
