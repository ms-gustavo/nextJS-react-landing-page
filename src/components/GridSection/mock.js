export default {
  background: false,
  title: 'My grid',
  description:
    'Lorem ipsum dolor sit, amet consectetur adipisicing elit. Provident harum quo esse cupiditate et ab optio debitis, est facilis in deleniti inventore vel aperiam cumque consequuntur nesciunt nemo qui necessitatibus.',
  grid: [
    {
      title: 'Teste 1',
      description:
        'Lorem ipsum dolor sit amet consectetur, adipisicing elit. Autem vitae unde at, iste quaerat, numquam ullam natus molestiae voluptatum dolorem quibusdam iusto reprehenderit maxime suscipit, accusantium quos? Ipsam pariatur fugit molestias perferendis nisi quis quaerat nemo, totam, quos, doloribus consectetur?',
    },
    {
      title: 'Teste 2',
      description:
        'Lorem ipsum dolor sit amet consectetur, adipisicing elit. Autem vitae unde at, iste quaerat, numquam ullam natus molestiae voluptatum dolorem quibusdam iusto reprehenderit maxime suscipit, accusantium quos? Ipsam pariatur fugit molestias perferendis nisi quis quaerat nemo, totam, quos, doloribus consectetur?',
    },
    {
      title: 'Teste 3',
      description:
        'Lorem ipsum dolor sit amet consectetur, adipisicing elit. Autem vitae unde at, iste quaerat, numquam ullam natus molestiae voluptatum dolorem quibusdam iusto reprehenderit maxime suscipit, accusantium quos? Ipsam pariatur fugit molestias perferendis nisi quis quaerat nemo, totam, quos, doloribus consectetur?',
    },
  ],
};
