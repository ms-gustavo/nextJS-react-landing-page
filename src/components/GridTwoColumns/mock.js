export default {
  title: 'Grid Two Columns',
  text: `Lorem ipsum dolor sit amet consectetur adipisicing elit. Distinctio fugit repudiandae dignissimos sequi delectus accusamus blanditiis, quisquam dolores esse libero reprehenderit enim? Fugiat optio labore distinctio asperiores quisquam suscipit ipsam?`,
  srcImg: 'assets/images/javascript.svg',
};
