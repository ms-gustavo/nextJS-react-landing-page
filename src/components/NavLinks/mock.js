export default [
  {
    children: 'Link 1',
    link: '#target 1',
    newTab: false,
  },
  {
    children: 'Link 2',
    link: '#target 2',
    newTab: false,
  },
  {
    children: 'Link 3',
    link: '#target 3',
    newTab: false,
  },
  {
    children: 'Link 4',
    link: '#target 4',
    newTab: false,
  },
  {
    children: 'Link 5',
    link: '#target 5',
    newTab: false,
  },
  {
    children: 'Link 6',
    link: '#target 6',
    newTab: false,
  },
  {
    children: 'Link 7',
    link: '#target 7',
    newTab: false,
  },
];
