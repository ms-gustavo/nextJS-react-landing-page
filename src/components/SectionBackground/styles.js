import styled, { css } from 'styled-components';

const containerBackgroundActivate = (theme) => css`
  background: ${theme.colors.primaryColor};
  color: ${theme.colors.white};
`;

export const Container = styled.div`
  ${({ theme, background }) => css`
    display: flex;
    align-items: center;
    background: ${theme.colors.white};
    color: ${theme.colors.primaryColor};

    ${background && containerBackgroundActivate(theme)};

    min-height: 100vh;
  `}
`;
