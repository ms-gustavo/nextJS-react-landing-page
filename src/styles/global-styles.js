import { createGlobalStyle, css } from 'styled-components';

export const GlobalStyles = createGlobalStyle`

  @import url('https://fonts.googleapis.com/css2?family=Montserrat:ital,wght@0,900;1,900&family=Open+Sans:wght@700&family=Poppins:ital,wght@0,300;1,300&display=swap');

*{
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

html {
  font-size: 62.5%;
}

body{
  font-size: 1.6rem;
  font-family: ${({ theme }) => theme.font.family.default}
}

h1, h2, h3, h4, h5, h6 {
  font-family: ${({ theme }) => theme.font.family.secondary}
}
`;
