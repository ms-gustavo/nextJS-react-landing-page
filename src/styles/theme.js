export const theme = {
  colors: {
    mainBg: 'purple',
    secondaryBg: 'yellow',
  },
  fonts: {},
  spacings: {},
};
