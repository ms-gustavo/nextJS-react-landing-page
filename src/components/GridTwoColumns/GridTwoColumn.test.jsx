import { screen } from '@testing-library/react';
import { GridTwoColumn } from '.';
import { renderTheme } from '../../styles/render-theme';

import mock from './mock';

describe('<GridTwoColumn/>', () => {
  it('should render grid with two columns', () => {
    const { container } = renderTheme(<GridTwoColumn {...mock} />);
    expect(screen.getByRole('heading')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
