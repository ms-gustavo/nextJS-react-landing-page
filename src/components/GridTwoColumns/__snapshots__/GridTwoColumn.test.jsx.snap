// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridTwoColumn/> should render grid with two columns 1`] = `
.c5 {
  color: #0A1128;
  font-size: 4rem;
  text-transform: uppercase;
}

.c2 {
  display: grid;
  grid-template-columns: 1fr 2fr;
  align-items: center;
  gap: 3.2rem;
}

.c2 .c4 {
  margin-bottom: 4rem;
}

.c0 {
  display: flex;
  align-items: center;
  min-height: 100vh;
  background: #FFFFFF;
  color: #0A1128;
}

.c1 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
}

.c6 {
  font-size: 2.4rem;
}

@media (max-width:768px) {
  .c2 {
    grid-template-columns: 1fr;
    text-align: center;
  }
}

@media (max-width:768px) {
  .c3 {
    margin-bottom: 3.2rem;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <div
          class="c3"
        >
          <h2
            class="c4 c5"
          >
            Grid Two Columns
          </h2>
          <div
            class="c6"
          >
            Lorem ipsum dolor sit amet consectetur adipisicing elit. Distinctio fugit repudiandae dignissimos sequi delectus accusamus blanditiis, quisquam dolores esse libero reprehenderit enim? Fugiat optio labore distinctio asperiores quisquam suscipit ipsam?
          </div>
        </div>
        <div
          class=""
        >
          <img
            alt="Grid Two Columns"
            class=""
            src="assets/images/javascript.svg"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
